bitnami/gitea/templates/externaldb-secrets.yaml

```
@@ -3,7 +3,7 @@ Copyright Broadcom, Inc. All Rights Reserved.
 SPDX-License-Identifier: APACHE-2.0
 */}}
 
-{{- if not .Values.postgresql.enabled }}
+{{- if and (not .Values.postgresql.enabled) (and (not .Values.externalDatabase.existingSecret) .Values.externalDatabase.password) }}
```

I suggest this simpler version, there is no need to evaluate `.Values.externalDatabase.password` in this condition:

```suggestion
{{- if not (or .Values.postgresql.enabled .Values.externalDatabase.existingSecret) }}
```

---

Applied myself